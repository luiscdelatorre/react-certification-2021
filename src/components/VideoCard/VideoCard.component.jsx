import React from 'react';
import { IoWifi } from 'react-icons/io5';
import { Card, Header, Content, Tag, Title, Subtitle, Text } from './VideoCard.styles';

const VideoCard = ({ title, channel, date, thumbnail, liveBroadcastContent }) => {
  return (
    <Card>
      <Header>
        <img src={thumbnail} alt={title} />
      </Header>
      <Content>
        {liveBroadcastContent !== 'none' ? (
          <Tag>
            <IoWifi />
            {liveBroadcastContent}
          </Tag>
        ) : (
          ''
        )}
        <Title>{title}</Title>
        <Subtitle>{channel}</Subtitle>
        <Text>{date}</Text>
      </Content>
    </Card>
  );
};

export default VideoCard;
