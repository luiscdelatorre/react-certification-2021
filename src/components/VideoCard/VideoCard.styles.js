import styled from 'styled-components';

const Card = styled.div`
  background-color: ${(props) => props.theme.cardBackground};
  border-radius: ${(props) => props.theme.borderRadiusLg};
  overflow: hidden;
  width: 100%;
  height: 100%;
`;

const Header = styled.div`
  img {
    height: 15rem;
    width: 100%;
    object-position: center;
    object-fit: cover;
  }
`;
const Content = styled.div`
  text-align: left;
  padding: 1rem;
`;

const Tag = styled.h6`
  background-color: ${(props) => props.theme.tagBackground};
  color: ${(props) => props.theme.tagText};
  display: inline-block;
  padding: 0.5rem 1rem;
  border-radius: ${(props) => props.theme.borderRadiusSm};
  letter-spacing: 0;
  margin: 0;
  text-transform: capitalize;
  line-height: 1.4rem;
  svg {
    vertical-align: bottom;
    font-size: 1.5rem;
    margin-right: 0.2rem;
  }
`;

const Title = styled.h4`
  color: ${(props) => props.theme.cardTitle};
`;
const Subtitle = styled.h5`
  color: ${(props) => props.theme.cardSubtitle};
`;
const Text = styled.p`
  color: ${(props) => props.theme.cardText};
  font-size: 1.2rem;
  margin: 0;
`;

export { Card, Header, Content, Tag, Title, Subtitle, Text };
