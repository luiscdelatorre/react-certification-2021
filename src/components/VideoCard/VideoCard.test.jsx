import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import VideoCard from './VideoCard.component';
import searchResult from '../../mock/youtube-videos-mock.json';

describe('VideoCard component', () => {
  it('Should display mock data in component', () => {
    const video = searchResult.items[1];
    render(
      <VideoCard
        title={video.snippet.title}
        channel={video.snippet.channelTitle}
        date={video.snippet.publishTime}
        thumbnail={video.snippet.thumbnails.default.url}
        liveBroadcastContent={video.snippet.liveBroadcastContent}
      />
    );

    expect(screen.getByText(video.snippet.title).tagName).toBe('H4');
    expect(screen.getByText(video.snippet.channelTitle).tagName).toBe('H5');
    expect(screen.getByText(video.snippet.publishTime).tagName).toBe('P');
  });

  it('Should render conditional item for liveBroadcastContent', () => {
    const video = searchResult.items[0];

    render(
      <VideoCard
        title={video.snippet.title}
        channel={video.snippet.channelTitle}
        date={video.snippet.publishTime}
        thumbnail={video.snippet.thumbnails.default.url}
        liveBroadcastContent={video.snippet.liveBroadcastContent}
      />
    );

    expect(screen.getByText('upcoming')).toBeInTheDocument();
  });

  it('Validates VideoCard snapshot', () => {
    const video = searchResult.items[0];
    const component = renderer.create(
      <VideoCard
        title={video.snippet.title}
        channel={video.snippet.channelTitle}
        date={video.snippet.publishTime}
        thumbnail={video.snippet.thumbnails.default.url}
        liveBroadcastContent={video.snippet.liveBroadcastContent}
      />
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
