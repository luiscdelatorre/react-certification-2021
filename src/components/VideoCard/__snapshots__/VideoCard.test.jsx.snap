// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoCard component Validates VideoCard snapshot 1`] = `
<div
  className="sc-bdfBwQ ffiwFl"
>
  <div
    className="sc-gsTCUz htpNBE"
  >
    <img
      alt="Wizeline"
      src="https://yt3.ggpht.com/ytc/AAUvwnighSReQlmHl_S_vSfvnWBAG5Cw4A0YxtE0tm5OpQ=s88-c-k-c0xffffffff-no-rj-mo"
    />
  </div>
  <div
    className="sc-dlfnbm hgzdhR"
  >
    <h6
      className="sc-hKgILt QCDMw"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M346.65 304.3a136 136 0 00-180.71 0 21 21 0 1027.91 31.38 94 94 0 01124.89 0 21 21 0 0027.91-31.4z"
        />
        <path
          d="M256.28 183.7a221.47 221.47 0 00-151.8 59.92 21 21 0 1028.68 30.67 180.28 180.28 0 01246.24 0 21 21 0 1028.68-30.67 221.47 221.47 0 00-151.8-59.92z"
        />
        <path
          d="M462 175.86a309 309 0 00-411.44 0 21 21 0 1028 31.29 267 267 0 01355.43 0 21 21 0 0028-31.31z"
        />
        <circle
          cx="256.28"
          cy="393.41"
          r="32"
        />
      </svg>
      upcoming
    </h6>
    <h4
      className="sc-eCssSg egxaJx"
    >
      Wizeline
    </h4>
    <h5
      className="sc-jSgupP cqSvwc"
    >
      Wizeline
    </h5>
    <p
      className="sc-gKsewC eyHrSn"
    >
      2014-09-27T01:39:18Z
    </p>
  </div>
</div>
`;
